onmt/translate/beam.py

```diff
@@ -170,12 +177,14 @@ def get_hyp(self, timestep, k):
         """
         Walk back to construct the full hypothesis.
         """
-        hyp, attn = [], []
+        hyp, attn, copy = [], [], []
         for j in range(len(self.prev_ks[:timestep]) - 1, -1, -1):
             hyp.append(self.next_ys[j + 1][k])
             attn.append(self.attn[j][k])
+            if self.copy_attn:
+                copy.append(self.copy[j][k])
             k = self.prev_ks[j][k]
-        return hyp[::-1], torch.stack(attn[::-1])
+        return hyp[::-1], torch.stack(attn[::-1]), copy[::-1]
```

Comment:
Can you make this a dictionary instead? 3 returns is a lot

Reply:
It is a really helpful idea, which makes the code more general. If dictionary is returned, we need to make many changes:

onmt/translate/beam.py,   the function get_hyp
`return hyp[::-1], torch.stack(attn[::-1]), copy[::-1]`    ->
`return   {"hyp": hyp[::-1], "att": torch.stack(attn[::-1]), "cop": copy[::-1]}`

onmt/translate/translator.py,    the function _from_beam
`hyp, att, cop = b.get_hyp(times, k)`
`hyps.append(hyp)`
`attn.append(att)`
`copy.append(cop)`
->
`hyp_dict = b.get_hyp(times, k)`
`hyps.append(hyp_dict['hyp'])`
`hyps.append(hyp_dict['att'])`
`hyps.append(hyp_dict['cop'])`

onmt/translate/translator.py , the function advance
`hyp, _, _ = self.get_hyp(le - 1, j)`
->
`hyp_dict = self.get_hyp(le - 1, j)`
`hyp = hyp_dict['hyp']`

How do you think the above changes? It seems that using dictionary may introduce many changes, maybe we can use a dictionary when 4 or more are returned from get_hyp function?